internal/compose/files.go

```diff
@@ -21,62 +21,95 @@ import (
 	"sort"
 )
 
+func findAndSortDocketFiles(prefix, mode string) ([]string, error) {
+	infos, err := ioutil.ReadDir(".")
+	if err != nil {
+		return nil, fmt.Errorf("failed to read current dir: %w", err)
+	}
+
+	files := make([]string, len(infos))
+	for i, info := range infos {
+		if info.IsDir() {
+			continue
+		}
+		files[i] = info.Name()
+	}
+
+	fs := newFileSorter(prefix, mode)
+	fs.AddFiles(files)
+
+	return fs.Results(), nil
+}
+
 // Ordering:
 //   prefix.yaml
 //   prefix.mode.yaml
 //   prefix.mode.extra.yaml
-func filterAndSortFilenames(prefix, mode string, files []string) []string {
-	if prefix == "" || mode == "" {
-		panic("prefix and mode must not be blank!")
+type fileSorter struct {
+	prefixOnly        []string
+	prefixAndMode     []string
+	prefixModeAndMore []string
+
+	prefixOnlyPattern       *regexp.Regexp
+	modeAndMaybeMorePattern *regexp.Regexp
+}
+
+func newFileSorter(prefix, mode string) *fileSorter {
+	if prefix == "" {
+		panic("prefix must not be empty!")
+	}
+	if mode == "" {
+		panic("mode must not be empty!")
 	}
 
-	prefixOnly := []string{}
-	prefixAndMode := []string{}
-	prefixModeAndMore := []string{}
+	return &fileSorter{
+		prefixOnly:        nil,
+		prefixAndMode:     nil,
+		prefixModeAndMore: nil,
 
-	prefixOnlyPattern := regexp.MustCompile(
-		fmt.Sprintf(`^%s\.ya?ml$`, regexp.QuoteMeta(prefix)))
+		prefixOnlyPattern: regexp.MustCompile(fmt.Sprintf(`^%s\.ya?ml$`,
+			regexp.QuoteMeta(prefix))),
 
-	modeAndMaybeMorePattern := regexp.MustCompile(
-		fmt.Sprintf(`^%s\.%s\.(.+\.)?ya?ml$`, regexp.QuoteMeta(prefix), regexp.QuoteMeta(mode)))
+		modeAndMaybeMorePattern: regexp.MustCompile(fmt.Sprintf(`^%s\.%s\.(.+\.)?ya?ml$`,
+			regexp.QuoteMeta(prefix), regexp.QuoteMeta(mode))),
+	}
+}
 
+func (fs *fileSorter) AddFiles(files []string) {
```

Comment:
i'd probably not have this function, and just do this in-line, but i don't feel strongly about it.

Reply:
gocognit definitely incentivizes you avoid nesting `if`s and loops, so this kinda happened along the way. I like that the sequence in the top-level function is clearly create, add files, get results. I think I'll keep it this way for now, though I do agree that having a function that just does a `for` loop is on the silly side. 😄